src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+
+        if (isBlank(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+
+        final CustomObjectQuery<NonResolvedReferencesCustomObject> customObjectQuery = CustomObjectQuery
+                .of(NonResolvedReferencesCustomObject.class)
+                .withPredicates(buildCustomObjectKeysQueryPredicate(singleton(key)));
+
+        return syncOptions
+                .getCtpClient()
+                .execute(customObjectQuery)
+                .thenApply(PagedResult::head);
+    }
+
+    private QueryPredicate<CustomObject<NonResolvedReferencesCustomObject>> buildCustomObjectKeysQueryPredicate(
+        @Nonnull final Set<String> customObjectKeys) {
+        final List<String> keysSurroundedWithDoubleQuotes = customObjectKeys.stream()
+                .filter(StringUtils::isNotBlank)
+                .map(customObjectKey -> format("\"%s\"", customObjectKey))
+                .collect(Collectors.toList());
+        String keysQueryString = keysSurroundedWithDoubleQuotes.toString();
+        // Strip square brackets from list string. For example: ["key1", "key2"] -> "key1", "key2"
+        keysQueryString = keysQueryString.substring(1, keysQueryString.length() - 1);
+        return QueryPredicate.of(format("key in (%s)", keysQueryString));
+    }
+
+    @Nonnull
+    @Override
+    public CompletionStage<Optional<CustomObject<NonResolvedReferencesCustomObject>>>
+        createOrUpdateCustomObject(@Nonnull final CustomObjectDraft<NonResolvedReferencesCustomObject>
+                                           customObjectDraft) {
+
+        return syncOptions
+                .getCtpClient()
+                .execute(CustomObjectUpsertCommand.of(customObjectDraft))
+                .handle((resource, exception) -> {
+                    if (exception == null) {
+                        return Optional.of(resource);
+                    } else {
+                        syncOptions.applyErrorCallback(
+                                format(CREATE_FAILED, customObjectDraft.getKey(), exception.getMessage()), exception);
+                        return Optional.empty();
+                    }
+                });
+    }
+
+    @Nonnull
+    @Override
+    public CompletionStage<Optional<CustomObject<NonResolvedReferencesCustomObject>>>
+        deleteCustomObject(@Nonnull final CustomObject<NonResolvedReferencesCustomObject> customObject) {
+
+        return syncOptions
+                .getCtpClient()
+                .execute(CustomObjectDeleteCommand.of(customObject, NonResolvedReferencesCustomObject.class))
+                .handle((resource, exception) -> {
+                    if (exception == null) {
+                        return Optional.of(resource);
+                    } else {
+                        syncOptions.applyErrorCallback(
+                                format(CREATE_FAILED, customObject.getKey(), exception.getMessage()), exception);
```

Comment:
`CREATE_FAILED` is wrong here.

Reply:
7aab8777db23834065e24ebdf96cf42e286ba22e